### DISCUSSION TITLE
Can't find package releases for composer wp-fastest-cache on wpackagist

### DISCUSSION THREAD

**Discussion by mrgesco** [renovatebot/renovate#10219] [Request Help] [answered]
Hi,

It seems that Renovate fails to retrieve versions for one package on wpackagist

```
WARN: Custom datasources are not allowed for this datasource and will be ignored
       "datasource": "git-tags",
       "registryUrls": ["https://wpackagist.org", "https://packagist.org"]

DEBUG: Found no results from datasource that look like a version (repository=XXXXX, dependency=wpackagist-plugin/wp-fastest-cache)
       "result": {
         "releases": [],
         "homepage": "https://wordpress.org/plugins/wp-fastest-cache/",
         "sourceUrl": "https://plugins.svn.wordpress.org/wp-fastest-cache/",
         "registryUrl": "https://wpackagist.org"
       }
```
And this is my composer.json
```
{
  "name": "test/renovate-test",
  "type": "project",
  "license": "MIT",
  "description": "Test Renovate",
  "repositories": [
    {
      "type": "composer",
      "url": "https://wpackagist.org",
      "only": ["wpackagist-plugin/*", "wpackagist-theme/*"]
    }
  ],
  "require": {
    "php": ">=7.1",
    "wpackagist-plugin/wordfence":"^7.1",
    "wpackagist-plugin/wp-fastest-cache":"^0.8"
  },
  "require-dev": {
    "wpackagist-plugin/wp-mailhog-smtp":"^1.0"
  },
  "config": {
    "optimize-autoloader": true,
    "preferred-install": "dist"
  },
  "minimum-stability": "dev",
  "prefer-stable": true,
  "extra": {
    "installer-paths": {
      "web/plugins/{$name}/": ["type:wordpress-plugin"]
    }
  }
}
```
Have you any idea why it's happening?

Test repository on GitHub
https://github.com/mrgesco/renovate-test

Thanks for your help.

**Comment by rarkins** [comment]
Can you create a minimal reproduction repo on GitHub.com for this?

**Reply by HonkingGoose** [reply]
Read this document for a guide on creating a minimal reproduction:

- https://github.com/renovatebot/renovate/blob/main/docs/development/minimal-reproductions.md

**Reply by mrgesco** [reply]
This is the test repository on GitHub
https://github.com/mrgesco/renovate-test

**Reply by rarkins** [reply]
Does it require all dependencies to be included in order to reproduce the problem? https://github.com/mrgesco/renovate-test/blob/01b67acebb76984a04113400a0a0685ee66ac605/composer.json#L13-L20

**Reply by mrgesco** [reply]
I updated the repository with only the faulty dependency, same behavior

**Reply by rarkins** [reply]
Thanks and can you past in the `packageFiles with updates` result here?

**Comment by mrgesco** [comment]
```
DEBUG: packageFiles with updates
{
  "config": {
    "composer": [
      {
        "packageFile": "composer.json",
        "deps": [
          {
            "depType": "require",
            "depName": "wpackagist-plugin/wp-fastest-cache",
            "currentValue": "^0.8",
            "datasource": "packagist",
            "depIndex": 0,
            "warnings": [],
            "sourceUrl": "https://plugins.svn.wordpress.org/wp-fastest-cache/",
            "homepage": "https://wordpress.org/plugins/wp-fastest-cache/",
            "updates": []
          }
        ],
        "lockFiles": [
          "composer.lock"
        ],
        "registryUrls": [
          "https://wpackagist.org",
          "https://packagist.org"
        ],
        "constraints": {
          "composer": "1.*"
        },
        "managerData": {
          "composerJsonType": "project"
        }
      }
    ]
  }
}
```

**Reply by rarkins** [reply]
![image](https://user-images.githubusercontent.com/6311784/120308197-1d939a80-c2d4-11eb-8920-8f50a44a6cef.png)

Possibilities:
- Renovate can't look up from that datasource at all, or
- Renovate can't parse the 4-precison versions like `0.9.1.8` (they are not semver-compliant)
- Renovate thinks `^0.8` satisfies latest version `0.9.1.8`

**Reply by mrgesco** [reply]
After trying a few things, the problem is your second thought. Renovate doesn't support the 4-precision versions.

Have you any idea how I can try to instruct Renovate to parse this 4 digits with a specific configuration?

**Reply by rarkins** [reply]
We can support 4-precision versions with the `loose` versioning however it doesn't support *range* syntax like `^0.8`. If you were to pin that dependency to a fixed version (e.g. `0.8.0.0`) and then configure loose versioning like the following then it might work:

```json
{
  "packageRules": [{
    "matchPackageNames": ["wpackagist-plugin/wp-fastest-cache"],
    "versioning": "loose"
  }]
}
```

**Reply by mrgesco** [reply]
Perfect thank you for your help!

**Comment by mrgesco** [comment, marked as answer]
Problem was the 4-precision version (0.9.1.8) not semver compliant
